transport/src/main/java/io/netty/channel/AdaptiveRecvByteBufAllocator.java

```diff
@@ -164,13 +165,9 @@ public AdaptiveRecvByteBufAllocator() {
      * @param maximum  the inclusive upper bound of the expected buffer size
      */
     public AdaptiveRecvByteBufAllocator(int minimum, int initial, int maximum) {
-        checkPositive(minimum, "minimum");
-        if (initial < minimum) {
-            throw new IllegalArgumentException("initial: " + initial);
-        }
-        if (maximum < initial) {
-            throw new IllegalArgumentException("maximum: " + maximum);
-        }
+        // 1 <= minimum <= initial <= maximum.
+        checkClosedInterval(minimum, 1, initial, "minimum");
```

Comment:
I'd rather not use the methods in this case, since it isn't 'minimum' that may be wrong, it could be 'initial'. Yes, the pre-existing code behave similarly, but it would have been easy to fix.